.github/workflows/flaky-tests.yml

```diff
@@ -0,0 +1,56 @@
+name: Flaky Tests Detection
+
+on:
+  schedule:
+    - cron: "0 2 * * 1" # Run every Monday at 2 AM
+  workflow_dispatch:
+    inputs:
+      skip-posting-issues:
+        description: 'Skip creating/updating GitHub issues (preview mode)'
+        required: false
+        default: 'true'
+        type: boolean
+
+permissions:
+  contents: read
+  issues: write
+  id-token: write
+
+jobs:
+  analyze-flaky-tests:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: Retrieve Secrets from Vault
+        id: get-secrets
+        uses: grafana/shared-workflows/actions/get-vault-secrets@5d7e361bc7e0a183cde8afe9899fb7b596d2659b # get-vault-secrets-v1.2.0
+        with:
+          repo_secrets: |
+            LOKI_URL=ci/repo/grafana/mimir/flaky-tests-bot:loki-url
+            LOKI_USERNAME=ci/repo/grafana/mimir/flaky-tests-bot:loki-username
+            LOKI_PASSWORD=ci/repo/grafana/mimir/flaky-tests-bot:loki-password
+            APP_ID=mimir-github-bot:app_id
+            PRIVATE_KEY=mimir-github-bot:private_key
+
+      - name: Generate GitHub App Token
+        id: github-token
+        uses: actions/create-github-app-token@v1
```

Comment:
Probably needs a hash, right?

Reply:
we're not using it with a hash anywhere else, i'm not sure why and also not sure why zizmor isn't yelling at us